FEM/Output.cpp

```diff
@@ -1709,8 +1709,35 @@ double COutput::NODWritePLYDataTEC(int number)
 	m_msh->getPointsForInterpolationAlongPolyline(ply, interpolation_points);
 	m_msh->setMinEdgeLength(tmp_min_edge_length);
 
-	//   std::cout << "size of nodes_vector: " << nodes_vector.size() << ", size of old_nodes_vector: " <<
-	//   old_nodes_vector.size() << "\n";
+    if (m_pcs)
+        if (m_pcs->getProcessType() == FiniteElement::HEAT_TRANSPORT_BHE)
+        {
+            for (std::size_t j(0); j < nodes_vector.size(); j++)
+            {
+                tec_file << interpolation_points[j] << " ";
+                for (std::size_t k = 0; k < no_variables; k++)
+                {
+                    if (NodeIndex[k] < 2) // this is soil temperature Ts
+                    {
+                        // get the global node index
+                        std::size_t global_idx(0), idx_bhe(0);
+                        for (idx_bhe = 0; idx_bhe < vec_BHEs.size(); idx_bhe++)
+                        {
+                            if (vec_BHEs[idx_bhe]->get_name().find(geo_name) != std::string::npos)
```

Comment:
maybe better to use `compare()` instead of `find()`. `find()` does not check complete match of strings